apps/aechannel/src/aesc_fsm.erl

```diff
@@ -3279,19 +3284,25 @@ report(Tag, Info, D) ->
                                  , tag  => Tag
                                  , info => Info }, D).
 
-report_info(DoRpt, Msg, #data{client_connected = false}) ->
-    lager:debug("No client. DoRpt = ~p, Msg = ~p", [DoRpt, Msg]),
-    ok;
-report_info(DoRpt, Msg0, #data{role = Role, client = Client} = D) ->
+report_info(DoRpt, Msg0, #data{ role = Role
+                              , client_connected = Connected
+                              , client = Client} = D) ->
     if DoRpt ->
             Msg = rpt_message(Msg0, D),
             lager:debug("~p report_info(true, Client = ~p, Msg = ~p)",
                         [Role, Client, Msg]),
-            Client ! {?MODULE, self(), Msg};
+            maybe_send_rpt(Connected, Client, Msg),
+            log(rpt, maps:get(tag, Msg), Msg, D);
        true  ->
             lager:debug("~p report_info(~p, ~p)", [Role, DoRpt, Msg0]),
-            ok
-    end,
+            D
```

Comment:
This is to cause some funny bugs in my PR. Note to self and future reviewers 😄

Reply:
Yes, careful rebasing will surely be required